BugFix: use proper C Preprocessor form

SlySven opened the pull request:
Using a only potentially defined symbol directly in a preprocessor `#if` lights up as a warning in the QtCreator IDE when the symbol is NOT defined.

It is cleaner to use the `defined(...)` macro(?)...

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: use proper C Preprocessor form

Using a only potentially defined symbol directly in a preprocessor `#if`
lights up as a warning in the QtCreator IDE when the symbol is NOT defined.

It is cleaner to use the `defined(...)` macro(?)...

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented:
Sorry, I don't understand the significance of this change.

SlySven commented:
> Sorry, I don't understand the significance of this change.

![image](https://user-images.githubusercontent.com/6163092/129090713-508c07f8-6da1-425f-a512-2daeb9524644.png)

Doing `#if VARIABLE_THAT_MIGHT_BE_DEFINED` is not good - `#if defined(VARIABLE_THAT_MIGHT_BE_DEFINED)` is much better - and while `#ifdef VARIABLE_THAT_MIGHT_BE_DEFINED` is almost as good we have something like 23 of the latter to over 200 of the former in our **own** code currently so I feel justified in using the first of these two...

vadi2 reviewed (approved):
Ah ok, that checks out. Thx!

SlySven committed:
Tickle: make an insignificant change to trigger a new CI build

Something seems amiss with the previous commit on GitHub as it did not run
the GH actions... looking at https://www.githubstatus.com/ it seems there
were problems at that time.

Actually this commit pulls in some ancillary files related to building in a
docker setup.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>
